caffe-distri/src/main/cpp/CaffeNet.cpp

```diff
@@ -531,7 +531,10 @@ int CaffeNet<Dtype>::getInitIter(int solver_index) {
         return root_solver_->iter();
     else {
         CHECK(syncs_[solver_index]);
-        return syncs_[solver_index]->solver()->iter();
+	if (solver_index == 0)
```

Comment:
No need to add condition for CPU mode since syncs_ size either zero or one in that mode.